src/apps/dashboard/modal/ready-for-loan-modal/ready-to-loan-modal-content.tsx

```diff
@@ -0,0 +1,237 @@
+    useState<Reservation[]>();
+  const { data: physicalReservations } = useGetReservationsV2();
+  const { data: digitalReservations } =
+    useGetV1UserReservations<ReservationListResult>();
+  const [allSelectableReservations, setAllSelectableReservations] = useState<
+    {
+      [key: string]: string;
+    }[]
+  >();
+  const [selectedReservations, setSelectedReservations] = useState<
+    {
+      [key: string]: string;
+    }[]
+  >([]);
+  const { mutate: deletePhysicalReservation } = useDeleteReservations();
+  const { mutate: deleteDigitalReservation } =
+    useDeleteV1UserReservationsIdentifier();
+  useEffect(() => {
+    if (physicalReservations && !physicalReservationsReadyToLoan) {
+      const readyToLoan = getReadyForPickup(physicalReservations);
+      if (readyToLoan) {
+        setPhysicalReservationsReadyToLoan(readyToLoan);
+      }
+    }
+  }, [physicalReservations, physicalReservationsReadyToLoan, today]);
+
+  useEffect(() => {
+    if (digitalReservations && !digitalReservationsReadyToLoan) {
+      const { reservations } = digitalReservations;
+      if (reservations) {
+        const readyToLoan = reservations.filter((reservation) => {
+          const expiryDate = dayjs(reservation.expireDateUtc);
+          return expiryDate > today;
+        });
+        if (readyToLoan) {
+          setDigitalReservationsReadyToLoan(readyToLoan);
+        }
+      }
+    }
+  }, [digitalReservations, digitalReservationsReadyToLoan, today]);
+
+  useEffect(() => {
+    if (
+      physicalReservationsReadyToLoan &&
+      digitalReservationsReadyToLoan &&
+      !allSelectableReservations
+    ) {
+      const fausts = physicalReservationsReadyToLoan.reduce((acc, pr) => {
+        return { ...acc, [pr.recordId]: pr.reservationId };
+      }, {});
+      const identsObj = digitalReservationsReadyToLoan.reduce((acc, dr) => {
+        return { ...acc, [dr.identifier as string]: dr.identifier };
+      }, {});
+      const selectableReservations = { ...fausts, ...identsObj };
+      if (Object.keys(selectableReservations).length > 0) {
+        setAllSelectableReservations(
+          selectableReservations as {
+            [key: string]: string;
+          }[]
+        );
+      }
+    }
+  }, [
+    allSelectableReservations,
+    physicalReservationsReadyToLoan,
+    digitalReservationsReadyToLoan
+  ]);
+
+  const selectAllQueuedResevationsHandler = () => {
+    if (Object.keys(selectedReservations).length > 0) {
+      setSelectedReservations([]);
+    } else if (allSelectableReservations) {
+      setSelectedReservations(allSelectableReservations);
+    }
+  };
+  const setCustomSelection = useCallback(
+    (elementId: string, reservationId: string) => {
+      if (Object.keys(selectedReservations).includes(elementId as string)) {
+        const newSelection = { ...selectedReservations };
+        delete newSelection[elementId as never];
+        setSelectedReservations(newSelection);
```

Comment:
There is something about the type assertions here. It should be possible to avoid them. But anyways...